extensions/arc/src/ui/dialogs/connectControllerDialog.ts

```diff
@@ -134,18 +175,67 @@ export class ConnectToControllerDialog extends InitializingComponent {
 			// default info.name to the name of the controller instance if the user did not specify their own and to a pre-canned default if for some weird reason controller endpoint returned instanceName is also not a valid value
 			controllerModel.info.name = controllerModel.info.name || controllerModel.controllerConfig?.metadata.name || loc.defaultControllerName;
 		} catch (err) {
-			vscode.window.showErrorMessage(loc.connectToControllerFailed(this.urlInputBox.value, err));
+			this.dialog.message = {
+				text: loc.connectToControllerFailed(this.urlInputBox.value, err),
+				level: azdata.window.MessageLevel.Error
+			};
 			return false;
 		}
 		this._completionPromise.resolve({ controllerModel: controllerModel, password: this.passwordInputBox.value });
 		return true;
 	}
+}
+export class PasswordToControllerDialog extends ControllerDialogBase {
 
-	private handleCancel(): void {
-		this._completionPromise.resolve(undefined);
+	constructor(protected _treeDataProvider: AzureArcTreeDataProvider) {
+		super(_treeDataProvider, loc.passwordToController);
 	}
 
-	public waitForClose(): Promise<ConnectToControllerDialogModel | undefined> {
-		return this._completionPromise.promise;
+	protected fieldToFocusOn = () => this.passwordInputBox;
```

Comment:
Please don't use anonymous functions at the class level - they're generally just harder to read (especially when mixed in with normally defined function syntax)